src/OutputWriters/field_slicer.jl

```diff
@@ -0,0 +1,60 @@
+struct FieldSlicer{I, W}
+    indices :: I
+    with_halos :: W
+end
+
+FieldSlicer(; x=:, y=:, z=:, with_halos=false) = FieldSlicer((x, y, z), with_halos)
```

Comment:
I will make this change --- but note that some users might be familiar with `:` but not familiar with `Colon()`. Which is better for the docstring?

Reply:
I guess `:` has three uses: symbols, ranges, and `Colon()` while `Colon()` has one use so it might be clearer.

The docstring for `Colon()` is more relevant than the docstring for `:`

```
help?> :
search: : :: ?: >: <:

  (:)(start, [step], stop)

  Range operator. a:b constructs a range from a to b with a step size of 1 (a UnitRange) , and a:s:b is similar but uses a step size of s (a StepRange).

  : is also used in indexing to select whole dimensions and for Symbol literals, as in e.g. :hello.

  ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

  (:)(I::CartesianIndex, J::CartesianIndex)

  Construct CartesianIndices from two CartesianIndex.

  │ Julia 1.1
  │
  │  This method requires at least Julia 1.1.

  Examples
  ≡≡≡≡≡≡≡≡≡≡

  julia> I = CartesianIndex(2,1);
  
  julia> J = CartesianIndex(3,3);
  
  julia> I:J
  2×3 CartesianIndices{2,Tuple{UnitRange{Int64},UnitRange{Int64}}}:
   CartesianIndex(2, 1)  CartesianIndex(2, 2)  CartesianIndex(2, 3)
   CartesianIndex(3, 1)  CartesianIndex(3, 2)  CartesianIndex(3, 3)
```

```
help?> Colon()
  Colon()

  Colons (:) are used to signify indexing entire objects or dimensions at once.

  Very few operations are defined on Colons directly; instead they are converted by to_indices to an internal vector type (Base.Slice) to represent the collection of indices they span
  before being used.

  The singleton instance of Colon is also a function used to construct ranges; see :.
```

Reply:
True! There's even another use for `:` in the ternary operator. I'll use `Colon()`.